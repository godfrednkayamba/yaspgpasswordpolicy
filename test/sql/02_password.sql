LOAD 'yaspgpp';
--
--reset all settings
--
SET yaspgpp.username_min_length TO DEFAULT;
SET yaspgpp.username_min_special TO DEFAULT;
SET yaspgpp.username_min_upper TO DEFAULT;
SET yaspgpp.username_min_upper TO DEFAULT;
SET yaspgpp.username_min_digit TO DEFAULT;
SET yaspgpp.username_contain_password TO DEFAULT;
SET yaspgpp.username_ignore_case TO DEFAULT;
SET yaspgpp.username_contain TO DEFAULT;
SET yaspgpp.username_not_contain TO DEFAULT;
SET yaspgpp.username_min_repeat TO DEFAULT;
SET yaspgpp.password_min_length TO DEFAULT;
SET yaspgpp.password_min_special TO DEFAULT;
SET yaspgpp.password_min_upper TO DEFAULT;
SET yaspgpp.password_min_upper TO DEFAULT;
SET yaspgpp.password_min_digit TO DEFAULT;
SET yaspgpp.password_contain_username TO DEFAULT;
SET yaspgpp.password_ignore_case TO DEFAULT;
SET yaspgpp.password_contain TO DEFAULT;
SET yaspgpp.password_not_contain TO DEFAULT;
SET yaspgpp.password_min_repeat TO DEFAULT;
--password checks
--
--length must be >=2
--
SET yaspgpp.password_min_length TO 2;
DROP USER IF EXISTS aa;
CREATE USER aa WITH PASSWORD 'd';
CREATE USER aa WITH PASSWORD 'dd';
DROP USER IF EXISTS aa;

--
--min special >= 1
--
SET yaspgpp.password_min_special TO 1;
DROP USER IF EXISTS aa;
CREATE USER aa WITH PASSWORD 'aa';
CREATE USER aa WITH PASSWORD 'a$';
DROP USER IF EXISTS aa;
--
--min upper >=1
--
SET yaspgpp.password_min_upper TO 1;
DROP USER IF EXISTS "aa";
CREATE USER "aa" WITH PASSWORD 'aa$';
CREATE USER "aa" WITH PASSWORD 'aA$';
DROP USER IF EXISTS "aa";
--
--min lower >=2
--
SET yaspgpp.password_min_lower TO 1;
DROP USER IF EXISTS "aa";
CREATE USER "aa" WITH PASSWORD 'AA$';
CREATE USER "aa" WITH PASSWORD 'aA$';
DROP USER IF EXISTS "aa";
--
--must contain one of the characters 'a','b','c'
--
SET yaspgpp.password_contain TO 'a,b,c';
DROP USER IF EXISTS "aa";
CREATE USER "aa" WITH PASSWORD 'dddU$';
CREATE USER "aa" WITH PASSWORD 'ddaU$';
DROP USER IF EXISTS "aa";
--
--must not contain one of the characters 'x','z'
--
SET yaspgpp.password_not_contain TO 'x,z';
DROP USER IF EXISTS "aa";
CREATE USER "aa" WITH PASSWORD 'Ax$';
CREATE USER "aa" WITH PASSWORD 'Ab$';
DROP USER IF EXISTS "aa";
--
--passord contain username
--
SET yaspgpp.password_contain_username TO on;
DROP USER IF EXISTS "aa";
CREATE USER "aa" WITH PASSWORD 'aa$';
CREATE USER "aa" WITH PASSWORD 'Ab$';
DROP USER IF EXISTS "aa";
--
--ignore case while performing checks
--
SET yaspgpp.password_ignore_case TO on;
DROP USER IF EXISTS "aa";
CREATE USER "aa" WITH PASSWORD 'random_AA$';
DROP USER IF EXISTS "aa";
--
--min digit >=1
--
SET yaspgpp.password_min_digit TO 1;
DROP USER IF EXISTS aa;
CREATE USER aa WITH PASSWORD 'a@a';
CREATE USER aa WITH PASSWORD 'a@1';
DROP USER IF EXISTS aa;
--
--min password repeat 2
--
SET yaspgpp.password_min_repeat TO 2;
DROP USER IF EXISTS aa;
CREATE USER aa WITH PASSWORD '1a@bbb';
CREATE USER aa WITH PASSWORD '1a@a';
DROP USER IF EXISTS aa;
--
-- Check NULL password
--
CREATE USER aa WITH PASSWORD '1a@bcg';
ALTER USER aa PASSWORD NULL;
DROP USER IF EXISTS aa;
CREATE USER aa PASSWORD NULL;
DROP USER IF EXISTS aa;
--
-- Check whitlisted users
SET yaspgpp.password_min_repeat TO 2;
SET yaspgpp.whitelist = 'nocheck1,nocheck2,aaaaaaaa,bbbbbbbb,cccccccc,dddddddd,eeeeeeee,ffffffff,gggggggg';
DROP USER IF EXISTS nocheck1;
CREATE USER nocheck1 WITH PASSWORD 'aaaa';
DROP USER IF EXISTS nocheck1;
CREATE USER nocheck1;
DROP USER IF EXISTS nocheck2;
CREATE USER nocheck2 WITH PASSWORD 'aaaa';
ALTER USER nocheck2 WITH PASSWORD 'bbbb';
DROP USER IF EXISTS nocheck1;
DROP USER IF EXISTS nocheck2;


