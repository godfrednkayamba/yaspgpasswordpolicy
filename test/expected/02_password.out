LOAD 'yaspgpasswordpolicy';
--
--reset all settings
--
SET yaspgpasswordpolicy.username_min_length TO DEFAULT;
SET yaspgpasswordpolicy.username_min_special TO DEFAULT;
SET yaspgpasswordpolicy.username_min_upper TO DEFAULT;
SET yaspgpasswordpolicy.username_min_upper TO DEFAULT;
SET yaspgpasswordpolicy.username_min_digit TO DEFAULT;
SET yaspgpasswordpolicy.username_contain_password TO DEFAULT;
SET yaspgpasswordpolicy.username_ignore_case TO DEFAULT;
SET yaspgpasswordpolicy.username_contain TO DEFAULT;
SET yaspgpasswordpolicy.username_not_contain TO DEFAULT;
SET yaspgpasswordpolicy.username_min_repeat TO DEFAULT;
SET yaspgpasswordpolicy.password_min_length TO DEFAULT;
SET yaspgpasswordpolicy.password_min_special TO DEFAULT;
SET yaspgpasswordpolicy.password_min_upper TO DEFAULT;
SET yaspgpasswordpolicy.password_min_upper TO DEFAULT;
SET yaspgpasswordpolicy.password_min_digit TO DEFAULT;
SET yaspgpasswordpolicy.password_contain_username TO DEFAULT;
SET yaspgpasswordpolicy.password_ignore_case TO DEFAULT;
SET yaspgpasswordpolicy.password_contain TO DEFAULT;
SET yaspgpasswordpolicy.password_not_contain TO DEFAULT;
SET yaspgpasswordpolicy.password_min_repeat TO DEFAULT;
--password checks
--
--length must be >=2
--
SET yaspgpasswordpolicy.password_min_length TO 2;
DROP USER IF EXISTS aa;
NOTICE:  role "aa" does not exist, skipping
CREATE USER aa WITH PASSWORD 'd';
ERROR:  password length should match the configured yaspgpasswordpolicy.password_min_length (2)
CREATE USER aa WITH PASSWORD 'dd';
DROP USER IF EXISTS aa;
--
--min special >= 1
--
SET yaspgpasswordpolicy.password_min_special TO 1;
DROP USER IF EXISTS aa;
NOTICE:  role "aa" does not exist, skipping
CREATE USER aa WITH PASSWORD 'aa';
ERROR:  username should not contain password
CREATE USER aa WITH PASSWORD 'a$';
DROP USER IF EXISTS aa;
--
--min upper >=1
--
SET yaspgpasswordpolicy.password_min_upper TO 1;
DROP USER IF EXISTS "aa";
NOTICE:  role "aa" does not exist, skipping
CREATE USER "aa" WITH PASSWORD 'aa$';
ERROR:  password should not contain username
CREATE USER "aa" WITH PASSWORD 'aA$';
DROP USER IF EXISTS "aa";
--
--min lower >=2
--
SET yaspgpasswordpolicy.password_min_lower TO 1;
DROP USER IF EXISTS "aa";
NOTICE:  role "aa" does not exist, skipping
CREATE USER "aa" WITH PASSWORD 'AA$';
ERROR:  password does not contain the configured yaspgpasswordpolicy.password_min_lower characters (1)
CREATE USER "aa" WITH PASSWORD 'aA$';
DROP USER IF EXISTS "aa";
--
--must contain one of the characters 'a','b','c'
--
SET yaspgpasswordpolicy.password_contain TO 'a,b,c';
DROP USER IF EXISTS "aa";
NOTICE:  role "aa" does not exist, skipping
CREATE USER "aa" WITH PASSWORD 'dddU$';
ERROR:  password does not contain the configured yaspgpasswordpolicy.password_contain characters: a,b,c
CREATE USER "aa" WITH PASSWORD 'ddaU$';
DROP USER IF EXISTS "aa";
--
--must not contain one of the characters 'x','z'
--
SET yaspgpasswordpolicy.password_not_contain TO 'x,z';
DROP USER IF EXISTS "aa";
NOTICE:  role "aa" does not exist, skipping
CREATE USER "aa" WITH PASSWORD 'Ax$';
ERROR:  password does not contain the configured yaspgpasswordpolicy.password_contain characters: a,b,c
CREATE USER "aa" WITH PASSWORD 'Ab$';
DROP USER IF EXISTS "aa";
--
--passord contain username
--
SET yaspgpasswordpolicy.password_contain_username TO on;
DROP USER IF EXISTS "aa";
NOTICE:  role "aa" does not exist, skipping
CREATE USER "aa" WITH PASSWORD 'aa$';
ERROR:  password should not contain username
CREATE USER "aa" WITH PASSWORD 'Ab$';
DROP USER IF EXISTS "aa";
--
--ignore case while performing checks
--
SET yaspgpasswordpolicy.password_ignore_case TO on;
DROP USER IF EXISTS "aa";
NOTICE:  role "aa" does not exist, skipping
CREATE USER "aa" WITH PASSWORD 'random_AA$';
ERROR:  password should not contain username
DROP USER IF EXISTS "aa";
NOTICE:  role "aa" does not exist, skipping
--
--min digit >=1
--
SET yaspgpasswordpolicy.password_min_digit TO 1;
DROP USER IF EXISTS aa;
NOTICE:  role "aa" does not exist, skipping
CREATE USER aa WITH PASSWORD 'a@a';
ERROR:  password does not contain the configured yaspgpasswordpolicy.password_min_digit characters (1)
CREATE USER aa WITH PASSWORD 'a@1';
DROP USER IF EXISTS aa;
--
--min password repeat 2
--
SET yaspgpasswordpolicy.password_min_repeat TO 2;
DROP USER IF EXISTS aa;
NOTICE:  role "aa" does not exist, skipping
CREATE USER aa WITH PASSWORD '1a@bbb';
ERROR:  password characters are repeated more than the configured yaspgpasswordpolicy.password_min_repeat times (2)
CREATE USER aa WITH PASSWORD '1a@a';
DROP USER IF EXISTS aa;
--
-- Check NULL password
--
CREATE USER aa WITH PASSWORD '1a@bcg';
ALTER USER aa PASSWORD NULL;
DROP USER IF EXISTS aa;
CREATE USER aa PASSWORD NULL;
DROP USER IF EXISTS aa;
--
-- Check whitlisted users
SET yaspgpasswordpolicy.password_min_repeat TO 2;
SET yaspgpasswordpolicy.whitelist = 'nocheck1,nocheck2,aaaaaaaa,bbbbbbbb,cccccccc,dddddddd,eeeeeeee,ffffffff,gggggggg';
DROP USER IF EXISTS nocheck1;
NOTICE:  role "nocheck1" does not exist, skipping
CREATE USER nocheck1 WITH PASSWORD 'aaaa';
DROP USER IF EXISTS nocheck1;
CREATE USER nocheck1;
DROP USER IF EXISTS nocheck2;
NOTICE:  role "nocheck2" does not exist, skipping
CREATE USER nocheck2 WITH PASSWORD 'aaaa';
ALTER USER nocheck2 WITH PASSWORD 'bbbb';
DROP USER IF EXISTS nocheck1;
DROP USER IF EXISTS nocheck2;
